game.py

```diff
@@ -0,0 +1,302 @@
+        new_game_state._board[move[0]][move[1]] = new_game_state._player_next
+        new_game_state._player_next = - new_game_state._player_next
+        return new_game_state
+        
+    def unmake_move(self):
+        new_game_state = self._previous_game_state
+        return new_game_state
+        
+    def get_next_moves(self, moves_sort = False, game_end = False):
+        moves = []
+        any_moves = False
+        for i in range(len(self._board)):
+            if any_moves:
+                break
+            for j in range(len(self._board[i])):
+                if self._board[i][j] == None:
+        #            moves.append ((i, j))
+                    moves.insert(0, (i, j))
+                    if game_end:
+                        any_moves = True
+                        break
+                    
+        random.shuffle(moves)
+        
+        if moves_sort:
+            moves_and_scores = []
+            i = 0
+            for move in moves:
+                new_game_state = self.make_move(move)
+                moves_and_scores.append([move])
+                moves_and_scores[i].append(new_game_state.score())
+                i = i + 1
+            print ('Moves and scores before sort : ', moves_and_scores)
+            moves_and_scores.sort(key = lambda x: x[1])
+            print ('Moves and scores after sort : ', moves_and_scores)
+            moves = []
+            for move_and_score in moves_and_scores:
+                moves.append(move_and_score[0])
+                
+        # print('moves are ', moves)
+        return moves
+
+    
+
+
+    def game_end(self):
+        if self.get_next_moves(game_end = True) == []:
+            return True
+        else:
+            return False
+
+    def valid_move(self, move):
+        if move in self.get_next_moves():
+            return True
+        else:
+            return False
+   
+        
+    def score(self):
+        score = 0
+        for i in range(len(self._board)):
+            for j in range(len(self._board[i])):
+                if self._board[i][j] == self._player_next:
+                    score = score + (i + 1)*(j + 1)
+                else:
+                    if self._board[i][j] == - self._player_next:
+                        score = score - (i + 1)*(j + 1)
+
+        global score_count
+        score_count = score_count + 1
+
+        ## print('score = ', score)
+        return score
+
+def minimax(game_state, look_forward,  player_next_first, alpha, beta, sort_depth):
+ 
+    # print('look forward = ', look_forward)
+    move_and_score = ((-1, -1), 0)
+    
+    if look_forward == 0 or game_state.game_end():
+        move_score = game_state.score()
+        if game_state.get_player_next() != player_next_first: 
+            move_score = - move_score
+               
+        ## print('move score = ', move_score, 'Game End : ', game_state.game_end())
+        return (-1, -1), move_score
+
+    if sort_depth <= look_forward:

```

Comment:
I don't think `sort_depth` is currently working as advertised, I think it's a bit inverted. For example, with `sort_depth` of 1 I think that currently means we will always sort at every level of the tree no matter how far our `look_forward` is.